src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -325,22 +325,22 @@ std::optional< homestore::chunk_num_t > HSHomeObject::get_shard_chunk(shard_id_t
     return std::make_optional< homestore::chunk_num_t >(hs_shard->sb_->chunk_id);
 }
 
-std::optional< homestore::chunk_num_t > HSHomeObject::get_any_chunk_id(pg_id_t const pg_id) {
+std::tuple< bool, bool, homestore::chunk_num_t > HSHomeObject::get_any_chunk_id(pg_id_t pg_id) {
     std::scoped_lock lock_guard(_pg_lock);
     auto pg_iter = _pg_map.find(pg_id);
-    RELEASE_ASSERT(pg_iter != _pg_map.end(), "Missing PG info");
+    if (pg_iter == _pg_map.end()) { return {false /* pg_found */, false /* shards_found */, 0 /* chunk_id */}; }
+
     HS_PG* pg = static_cast< HS_PG* >(pg_iter->second.get());
-    if (pg->any_allocated_chunk_id_.has_value()) {
-        // it is already cached and use it;
-        return pg->any_allocated_chunk_id_;
+    if (pg->any_allocated_chunk_id_.has_value()) { // it is already cached and use it;
+        return {true /* pg_found */, true /* shards_found */, *pg->any_allocated_chunk_id_};
     }
 
     auto& shards = pg->shards_;
-    if (shards.empty()) { return std::nullopt; }
+    if (shards.empty()) { return {true /* pg_found */, false /* shards_found */, 0 /* chunk_id */}; }
+
     auto hs_shard = d_cast< HS_Shard* >(shards.front().get());
```

Comment:
this function is used to select a chunk to put shard_info_super_blk when creating a shard. not quite understand why to select the chunk of the front shard in the pg shard list. should we select a chunk through chunkselector for a newly created shard?

Reply:
ok, i think we need to revisit the current logic here. i doubt this probably casued the io_uring error we met before when creating shard.